### PR TITLE
Add CC13xx / CC26xx I2C support

### PR DESCRIPTION
**Pull request by bwitherspoon**
Add support for I2C on CC13xx / CC26xx SoCs and boards. This driver is functional, but is still a work in progress until I receive new hardware for testing. It does not yet fully comply with the I2C API.

### CODE REVIEW

**Review by vanti — CHANGES_REQUESTED** [review]

**Comment by bwitherspoon** [comment]
If you have a BOOSTXL-SENSORS expansion some basic testing can be done with https://github.com/bwitherspoon/zephyr/tree/boostxl_sensors and `west build -b cc1352r1_launchxl zephyr/samples/sensor/bme280/ -- -DSHIELD=boostxl_sensors`.

File: drivers/i2c/i2c_cc13xx_cc26xx.c [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+		I2CMasterControl(base, I2C_MASTER_CMD_SINGLE_SEND);
+
+		k_sem_take(&data->complete, K_FOREVER);
+
+		return data->error == I2C_MASTER_ERR_NONE ? 0 : -EIO;
+	}
+
+	/* Burst transmission */
+	I2CMasterDataPut(base, buf[0]);
+
```
[the remark this answers is not in the document]
Comment (bwitherspoon) [reply drivers/i2c/i2c_cc13xx_cc26xx.c]: Added a comment and a note for someone wanting to implement multiple master support.

File: drivers/i2c/i2c_cc13xx_cc26xx.c [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+	}
+
+	I2CMasterDataPut(base, buf[len - 1]);
+
+	I2CMasterControl(base, I2C_MASTER_CMD_BURST_SEND_FINISH);
+
+	k_sem_take(&data->complete, K_FOREVER);
+
+	if (data->error != I2C_MASTER_ERR_NONE) {
+		goto send_error_stop;
```
[the remark this answers is not in the document]
Comment (bwitherspoon) [reply drivers/i2c/i2c_cc13xx_cc26xx.c]: I see. That makes sense. Fixed.

File: drivers/i2c/i2c_cc13xx_cc26xx.c [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+{
+	u32_t cfg;
+	int err;
+
+	/* Enable I2C power domain */
+	PRCMPowerDomainOn(PRCM_DOMAIN_SERIAL);
+
+	/* Enable I2C peripheral clock */
+	PRCMPeripheralRunEnable(PRCM_PERIPH_I2C0);
+
```
[the remark this answers is not in the document]
Comment (bwitherspoon) [reply drivers/i2c/i2c_cc13xx_cc26xx.c]: Done. Same in other drivers.

**Comment by bwitherspoon** [comment]
I updated the boards' YAML and documentation to indicate support for I2C.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Review by vanti — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**